src/libicalvcard/vcardcomponent.c

```diff
@@ -1044,6 +1047,7 @@ static void comp_to_v3(vcardcomponent *impl)
         vcardvalue *value = vcardproperty_get_value(prop);
         vcardvalue_kind vkind = vcardvalue_isa(value);
         vcardparameter *param;
+        vcardenumarray_element type;
         char *subtype = NULL, *buf = NULL;
```

Comment:
Let's just move this declaration outside the loop and we can use it throughout the function, including for the TYPE=PREF loop